subsys/bluetooth/mesh/net.c

```diff
@@ -530,12 +530,13 @@ int bt_mesh_net_send(struct bt_mesh_net_tx *tx, struct net_buf *buf,
 		goto done;
 	}
 
+	BT_MESH_ADV(buf)->cb = cb;
+	BT_MESH_ADV(buf)->cb_data = cb_data;
```

Comment:
This doesn't seem like it should be the network layer's responsibility, but we can address this later.